src/layout/core/borders/container_draw.rs

```diff
@@ -33,21 +33,96 @@ impl ContainerDraw {
         x = gap / 2.0;
         w -= gap;
 
-        // Draw background of title bar
-        self.base.set_source_rgb(1.0, 0.0, 0.0);
-        self.base.set_color_source(title_color);
-        self.base.rectangle(x, 0.0, w, title_size);
-        self.base = try!(self.base.check_cairo());
-        self.base.fill();
-        self.base = try!(self.base.check_cairo());
-
-        // Draw title text
-        self.base.move_to(title_x, title_y);
-        self.base = try!(self.base.check_cairo());
-        self.base.set_color_source(title_font_color);
-        self.base = try!(self.base.check_cairo());
-        self.base.show_text(title.as_str());
-        self.base = try!(self.base.check_cairo());
+        // TODO: This shouldn't be clone for performance reasons, but self
+        // needs to not be borrowed later with all the paint operations, and
+        // I didn't know how to work around that.
+        let children = self.base.inner().children.clone();
+        let layout = self.base.inner().layout;
+
+        match (layout, children) {
+            (Some(Layout::Tabbed), Some(children)) => {
+                let tab_width = w / children.titles.len() as f64;
+
+                for (i, title) in children.titles.iter().enumerate() {
+                    let active = children.index == i;
+
+                    let title_color = if active { title_color }
+                        else { Borders::default_title_color() };
+
+                    let title_font_color = if active { title_font_color }
+                        else { Borders::default_title_font_color() };
+
+                    let x = x + tab_width*i as f64;
```

Comment:
Hmm, it took me a couple of moments to realize you are rebinding due to the loop and you need the previous `x` value.

I would suggest maybe trying to make a different name, as this looks confusing and very easy to introduce a bug by removing this "useless" looking rebinding.

If you did this by copying me, then this was a bad mistake I have done and I will endeavour not to make this same mistake since it's obviously rather confusing.

Reply:
I don't think rebindings are that hard to tell, I did notice yours and didn't have difficulty understanding what they were, and I also use them a lot, but I don't really know if they are generally considered good or bad practice.

Reply:
Ok, if you found it easy to follow for my previous code  it's probably not a big deal. 

Rebinding is generally frowned upon but it is context sensitive (eg I do it quite often when I need to eg turns Rust string to a CString but I want to take in a Rust string for ergonomics. There it's pretty clear, here it's only a little more dubious)